.travis.yml

```diff
@@ -1,45 +1,86 @@
+    - TARGET=license
 
-before_install:
-  - docker login -u=$DOCKER_USERNAME -p=$DOCKER_PASSWORD
-  - sudo rm /usr/local/bin/docker-compose
-  - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
-  - chmod +x docker-compose
-  - sudo mv docker-compose /usr/local/bin
+matrix:
+  fast_finish: true
 
 install:
-  - make start
+  - ./scripts/install_gcloud.sh
+
+before_script:
+  - echo ${DOCKER_PASSWORD} | docker login -u=decaftravis --password-stdin
+  - make setup
 
 script:
-  - make qa
-  - docker-compose logs
+  - docker pull ${IMAGE}
+  - make ${TARGET}
 
-after_success:
-  - docker-compose exec web /bin/bash -c "codecov --token $CODECOV_TOKEN"
+stages:
+  - build
+  - test
+  - push-tags
+  - name: deploy-staging
+    if: branch = devel
+  - name: deploy-production
+    if: branch = master
 
-deploy:
-  provider: script
-  script: ./push_to_hub_travis.sh
-  on:
-    all_branches: true
+jobs:
+  include:
+    - stage: build
+      install:
+        - ./scripts/install_gcloud.sh
+        - echo ${DOCKER_PASSWORD} | docker login -u=decaftravis --password-stdin
+      before_script:
+        - docker build --build-arg PIPENV_FLAGS="--deploy" -t ${IMAGE} .
+        - docker push ${IMAGE}
+      script: skip
+    - stage: push-tags
+      install:
+        - ./scripts/install_gcloud.sh
+      before_script:
+        - docker pull ${IMAGE}
+        - docker tag ${IMAGE} ${IMAGE_REPO}:${TRAVIS_COMMIT::12}
+        - docker tag ${IMAGE} ${IMAGE_REPO}:${TRAVIS_BRANCH}
+        - docker push ${IMAGE_REPO}:${TRAVIS_COMMIT::12}
+        - docker push ${IMAGE_REPO}:${TRAVIS_BRANCH}
+      script: skip
+    - stage: deploy-staging
+      install:
+        - ./scripts/install_kubectl.sh
+      script: skip
+      deploy:
+        provider: script
+        script: ./scripts/travis_deploy.sh model-staging
+    - stage: deploy-production
+      install:
+        - ./scripts/install_kubectl.sh
+      script: skip
+      deploy:
+        provider: script
+        script: ./scripts/travis_deploy.sh model-production
 
 notifications:
+  email: false
   slack:
-    on_success: never
     rooms:
-      - secure: GFboHh9KsjR5earrVk2ddz+ncxDYl53x22R0nvhBflSUjK1IY7kbM/ReUJIpd4JYgx+PSnk82gX/VrJ8ZmuiDSjQX0QgTJDaGB43bSTo0RrD8xBRGBQG5+DIh8YG51ZEZ2tHg91pJ4VJ1PJOs0+bYDDceyH4Ih5Ef2XMrdBzIzYVoegAm1RdjaXu9gp/C+ULc6JOnrudQxc2iEYo28QywYSQaYE+S/7J9LJ6rcfFmPNMvvEOfQJWreapnGALDT50PvtW4NrlkI6VJePRbLP2mLYC6Gva6HUY8Vys8bM6Xs+IZNKA4KRbWbVgsKfNTfySdGDiO+hsaBMEuM1IdsZkweAXZs/mQSr9mEtDgbeUkhyl3M0DS8ER/rs+XH13xW9Re7km73kmKdyPnjc6qJ7+io4521m6+XDYXxDrxMpTcfrTnqnPAAKAm2wRcddlFUJ6x6Nc+cGS0VvsQKjnU97sZFNo6iAAvIyiJ+QLNvRhQRDgAaMS8PF84UkrIlAvx4SeruMF+zf4EI9f1OtePklOrJPkYGgoV4ANEkfD4ez4dUqGLfppUMq70tIMRiLUWXRiLcyYc9P0mAW8NWLGw2Oi3+soRqOv/IzWHIZmUlTTjHfdESZPsCXYAemmhWmaSLtqBjIHCHM3YoXO1JTgqRW1mCA3/ronrfucGGgBzExxuf8=
+      - ${SLACK_ACCOUNT}:${SLACK_TOKEN}#${SLACK_CHANNEL}
+    on_success: never
```

Comment:
`change` would be more helpful IMO

Reply:
I agree with that, at least for `master` and `devel`. Unfortunately there is no way to limit notifications to specific branches on Travis, but let's try enabling this and see if it becomes too noisy.